test/test_smem_reuse.cpp

```diff
@@ -415,4 +417,90 @@ TEST_F(SmemReuseTest, MultiplePromoteReuse) {
   }
 }
 
+// Test that intervals with "skipped" positions in alias analysis are not
+// ignored This can happen when we require a sync immediately between two
+// https://github.com/NVIDIA/Fuser/pull/1381
+TEST_F(SmemReuseTest, SkippedSyncInterval) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  int64_t D = 5;
+  int64_t H = 6;
+  int64_t W = 7;
+
+  auto tv0 = full(
+      {
+          IrBuilder::create<Val>(D),
+          IrBuilder::create<Val>(H),
+          IrBuilder::create<Val>(W),
+      },
+      fusion->oneVal(),
+      DataType::Float);
+  tv0->setMemoryType(MemoryType::Shared);
+
+  auto tv1 = neg(tv0); // generate a loop nest, which is the last use of tv0
+
+  // Write to tv2 in adjacent loop nest. Pad so we can't do inner aliasing.
+  auto tv2 = pad(tv1, {fusion->zeroVal(), fusion->oneVal()});
+  tv2->setMemoryType(MemoryType::Shared);
+
+  auto tv3 = set(tv2); // third loop nest writes to global output
+
+  fusion->addOutput(tv3);
+
+  // By parallelizing the first loop, we remove its appearance in the kernel.
+  //
+  //   6  FOR threadIdx.x in ithreadIdx.x3{5}:
+  //   7    FOR i85 in iS4{6}:
+  //   8      FOR i86 in iS5{7}:
+  //   9        T0_s[ ithreadIdx.x0{5}, iS1{6}, iS2{7} ] ca_pos( 3 )
+  //      = full({5, 6, 7}, ( (float)(1) ));
+  //   10        T1_l[ ithreadIdx.x3{5}, iS4{6}, iS5{7} ] produce_pos( 3 )
+  //      = -T0_s[ ithreadIdx.x0{5}, iS1{6}, iS2{7} ] ca_pos( 3 );
+  //   14  FOR threadIdx.x in ithreadIdx.x6{5}:
```

Comment:
Why this is 14 instead of 11? Is it because there are three `END` for the three for loop? Could you explicitly write them out for clarity?

Reply:
Yes you got it. Good idea, I will add in the "endfor" positions for clarity.